openmdao/core/driver.py

```diff
@@ -655,14 +665,37 @@ def add_constraint(self, name, lower=None, upper=None, equals=None,
         if isinstance(equals, np.ndarray):
             equals = equals.flatten()
 
-        # Scale the lower and upper values
-        if lower is not None:
-            lower = (lower + adder)*scaler
-        if upper is not None:
-            upper = (upper + adder)*scaler
+        # Scale the equality bounds
         if equals is not None:
             equals = (equals + adder)*scaler
 
+        # Scale the upper and lower bounds unless they are infinite.
+        if lower is None:
+            pass
+        elif isinstance(lower, np.ndarray):
+            lower = lower.flatten()
+            # lower is an array, only scale/offset indices > -inf_bound
+            lower = np.where( lower > -inf_bound, (lower + adder)*scaler, -inf_bound )
```

Comment:
I vaguely remember this problem from before, and this doesn't seem like a good solution. If I have an N wide vector, and I want to constraint the 1st and last point, I end up with (n-2) extra infinite constraints that the optimizer thinks are real constraints, and wants derivatives for them (unless it can cull them somehow). I would think that taking the first and last values of the array and hooking them into the inputs of dummy pass-thru components, and then taking the output of those as the constraint would be better than this.